transaction/std/src/transaction_builder.rs

```diff
@@ -143,6 +159,26 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
         recipient: &PublicAddress,
         rng: &mut RNG,
     ) -> Result<(TxOut, TxOutConfirmationNumber), TxBuilderError> {
+        let context = self.add_output_with_context(value, recipient, rng)?;
+        Ok((context.tx_out, context.confirmation))
+    }
+
+    /// Add a non-change output to the transaction.
+    ///
+    /// If a sender memo credential has been set, this will create an
+    /// authenticated sender memo for the TxOut. Otherwise the memo will be
+    /// unused.
+    ///
+    /// # Arguments
+    /// * `value` - The value of this output, in picoMOB.
+    /// * `recipient` - The recipient's public address
+    /// * `rng` - RNG used to generate blinding for commitment
```

Comment:
The documentation is exactly the same here as the preceeding method which is confusing given they're both public. Even if the differences are subtle, they should be explained

Reply:
I'm wondering if it makes sense to keep both at all. I believe existing code could be changed to utilize struct deconstruction, e.g. `let TxOutContext { tx_out, confirmation, .. } = tx_builder.add_output(...);`

Reply:
Yeah I agree that this would be better

Reply:
Yeah if we're okay passing the shared secret as an output argument, this would be the better way